### PR TITLE
AVRO-3411: Support VSCode devcontainer

### PR DESCRIPTION
**Pull request by zcsizmadia**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO-3411)
  - https://issues.apache.org/jira/browse/AVRO-3411
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [X] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Set default behavior to automatically normalize line endings.
```
Comment (iemejia) [inline .gitattributes:1]: Do we really need this? I don't remember where we are towards line endings. @RyanSkraba maybe?

Comment (iemejia) [inline .gitattributes:1]: Do we really need this? Specially all the 'auto', isn't auto the default?
Should we cover the formats we don't care about e.g. doc/pdf/rtf?
I don't remember where we are towards line endings. @RyanSkraba maybe?

**Review by iemejia — COMMENTED** [review]
Thanks for bringing this one in @zcsizmadia !
I let some minor comments.

File: .gitattributes (hunk 1 above)
Comment (RyanSkraba) [reply .gitattributes:1]: There was a discussion on this about AVRO-3150 -- some tests were failing due to an overzealous checkstyle check IIRC.

Is there something that fails if this `.gitattributes` isn't present?   I'm not familiar with all the attributes, I admit!  What does `diff` and `astextplain` do? 

I have no objection to enforcing that script files are CRLF/LF no matter where they're checked out!

**Comment by zcsizmadia** [comment]
I heavily use devcontainer for other multi language/multi platform projects. AFAIK VSCode is the most used editor for such projects.

I like the fact that e.g. when I developed the extra codecs for C# , I could just generate the interop data for all the other languages, so my C# tests were running againsta ll  the compression codecs generated by the other languages.
Just simply ran ./build.sh interop...generate, without knowing anything about Ruby, PHP, or Java.

**Comment by zcsizmadia** [comment]
It is possible to have devcontainer config for each languages seperately, however since the dockerfile has all the tools and all the project depend on each other via the interop tests, I thought having full container setup for the whole project (based on the already existing dockerfile) makes more sense

**Review by iemejia — APPROVED** [review]
LGTM

I validated it in a local container on mac and it works.

I had some issues with the C# project import but we can fix those afterwards, merging this is great for the benefit of everyone who can now contribute in an easy reproducile way. Excellent work @zcsizmadia !

**Pull request merged**